hr_holidays_public/models/hr_leave.py

```diff
@@ -25,10 +25,18 @@ def _get_number_of_days(self, date_from, date_to, employee_id):
 
     @api.depends("number_of_days")
     def _compute_number_of_hours_display(self):
-        if self.holiday_status_id.exclude_public_holidays or not self.holiday_status_id:
-            instance = self.with_context(
-                exclude_public_holidays=True, employee_id=self.employee_id.id
+        """If the leave is validated, no call to `_get_number_of_days` is done, so we
+        need to inject the context here for including the public holidays if applicable.
+
+        For such cases, we need to serialize the call to super in fragments.
+        """
+        to_serialize = self.filtered(
+            lambda x: x.state == "validate"
```

Comment:
@pedrobaeza 

Did you remember why you filter on state ? It make the displayed value in the form false (when we create a new leave)

You can check here (today is public holiday ! :p )
http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

When I remove this filter it's ok but I cannot undestand what it is for

Reply:
The explanation is in the method docstring.

Reply:
I saw, but has I understood how it work we always need to push exclude_public_holidays=True in the context to get the good number of hours whatever the state

Reply:
Not on v13 at least, and not if the holidays type doesn't have the check.

Reply:
Today is defined has public holiday
![image](https://user-images.githubusercontent.com/6648433/155359895-cdaa10e4-0dbe-4b9d-b028-9ecb3a43af23.png)

"Compensatory days" type exclude public holidays
![image](https://user-images.githubusercontent.com/6648433/155360640-4d7ec1c6-a217-4d46-aa40-c8d44faaed8e.png)

This is not taken into account in the form view
![image](https://user-images.githubusercontent.com/6648433/155360187-f24b810e-e805-4466-a576-f0671bc22271.png)

From http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

I have the same behavor on 12.0. Remove state filter fix the problem

I'm really sorry if there's something I don't understand and I'm wasting your time but I feel like it's not working the way it should

Reply:
OK, then remove it such condition. Have you seen if in v13 is happening the same problem?

Reply:
That's the question. I don't think there's an issue in v13. The problem is here on v12.

Reply:
The screenshots are from the link above (13.0 runboat instance)